Intersect.Client/Interface/Game/Map/MinimapWindow.cs

```diff
@@ -0,0 +1,571 @@
+using Intersect.Client.Core;
+using Intersect.Client.Entities;
+using Intersect.Client.Framework.File_Management;
+using Intersect.Client.Framework.Graphics;
+using Intersect.Client.Framework.Gwen.Control;
+using Intersect.Client.Framework.Gwen.Control.EventArguments;
+using Intersect.Client.General;
+using Intersect.Client.Localization;
+using Intersect.Client.Maps;
+using Intersect.Enums;
+using Intersect.GameObjects;
+using Intersect.GameObjects.Maps;
+
+namespace Intersect.Client.Interface.Game.Map
+{
+    public sealed partial class MinimapWindow : Window
+    {
+        private GameRenderTexture _renderTexture;
+        private GameTexture _whiteTexture;
+
+        private bool _redrawMaps;
+        private bool _redrawEntities;
+        private byte _zoomLevel;
+        private Point _minimapTileSize;
+
+        private Dictionary<MapPosition, MapBase> _mapGrid = new();
+        private Dictionary<MapPosition, Dictionary<Point, EntityInfo>> _entityInfoCache = new();
+
+        private readonly Dictionary<MapPosition, GameRenderTexture> _minimapCache = new();
+        private readonly Dictionary<MapPosition, GameRenderTexture> _entityCache = new();
+        private readonly Dictionary<Guid, MapPosition> _mapPosition = new();
+        private readonly ImagePanel _minimap;
+        private readonly Button _zoomInButton;
+        private readonly Button _zoomOutButton;
+
+        private static readonly GameContentManager ContentManager = Globals.ContentManager;
+
+        // Constructors
+        public MinimapWindow(Base parent) : base(parent, Strings.Minimap.Title, false, "MinimapWindow")
+        {
+            DisableResizing();
+            IsHidden = true;
+
+            _minimap = new ImagePanel(this, "MinimapContainer");
+            _zoomInButton = new Button(_minimap, "ZoomInButton");
+            _zoomOutButton = new Button(_minimap, "ZoomOutButton");
+
+            LoadJsonUi(GameContentManager.UI.InGame, Graphics.Renderer.GetResolutionString());
+            EnsureInitialized();
```

Comment:
You are _not_ supposed to manually call this. It is _only_ called by `Window` to not needlessly configure stuff before the window is ever visible.